crates/goose/src/scheduler.rs

```diff
@@ -619,23 +404,26 @@ impl Scheduler {
     }
 
     pub async fn remove_scheduled_job(&self, id: &str) -> Result<(), SchedulerError> {
-        let mut jobs_guard = self.jobs.lock().await;
-        if let Some((job_uuid, scheduled_job)) = jobs_guard.remove(id) {
-            self.internal_scheduler
-                .remove(&job_uuid)
-                .await
-                .map_err(|e| SchedulerError::SchedulerInternalError(e.to_string()))?;
-
-            let recipe_path = Path::new(&scheduled_job.source);
-            if recipe_path.exists() {
-                fs::remove_file(recipe_path).map_err(SchedulerError::StorageError)?;
+        let (job_uuid, recipe_path) = {
+            let mut jobs_guard = self.jobs.lock().await;
+            match jobs_guard.remove(id) {
+                Some((uuid, job)) => (uuid, job.source.clone()),
+                None => return Err(SchedulerError::JobNotFound(id.to_string())),
             }
+        };
+
+        self.tokio_scheduler
+            .remove(&job_uuid)
+            .await
+            .map_err(|e| SchedulerError::SchedulerInternalError(e.to_string()))?;
 
-            self.persist_jobs_to_storage_with_guard(&jobs_guard).await?;
-            Ok(())
-        } else {
-            Err(SchedulerError::JobNotFound(id.to_string()))
+        let path = Path::new(&recipe_path);
+        if path.exists() {
+            fs::remove_file(path)?;
         }
+
+        persist_jobs(&self.storage_path, &self.jobs).await?;
+        Ok(())
```

Comment:
so we have to remove it from three or four places but each might fail, do we want to not abort on the first fail?